server/shared/publishing/helpers.js

```diff
@@ -89,19 +95,21 @@ function unpublishActivity(repository, activity) {
   return getPublishedStructure(repository).then(spine => {
     const spineActivity = find(spine.structure, { id: activity.id });
     if (!spineActivity) return;
+
     const deleted = getSpineChildren(spine, activity).concat(spineActivity);
-    return Promise.map(deleted, it => {
-      const filenames = getContentContainerFilenames(it);
-      return Promise.map(filenames, filename => {
-        const key = `${getBaseUrl(repository.id, it.id)}/${filename}.json`;
-        return storage.deleteFile(key);
-      });
-    }).then(() => {
-      spine.structure = filter(spine.structure, ({ id }) => !find(deleted, { id }));
-      return saveSpine(spine)
-        .then(savedSpine => updateRepositoryCatalog(repository, savedSpine.publishedAt))
-        .then(() => activity.save());
-    });
+    return Promise
+      .map(deleted, it => {
+        const baseUrl = getBaseUrl(repository.id, it.id);
+        const filePaths = getContainerFilePaths(baseUrl, it.contentContainers);
+
```

Comment:
I would say that this space is not necessary since the flow between the lines is consistent and obvious